regression/Makefile

```diff
@@ -20,15 +20,37 @@ DIRS = ansi-c \
        test-script \
        # Empty last line
 
-# Check for the existence of $dir. Tests under goto-gcc cannot be run on
-# Windows, so appveyor.yml unlinks the entire directory under Windows.
+# Run all test directories in sequence
+.PHONY: test
 test:
 	@for dir in $(DIRS); do \
-	  if [ -d "$$dir" ]; then \
-	    $(MAKE) -C "$$dir" test || exit 1; \
-	  fi; \
+        $(MAKE) "test-$$dir" || exit 1; \
 	done;
 
+# Pattern to execute a single directory, checking for existence of the directory
+# first. Tests under goto-gcc cannot be run on Windows, so appveyor.yml unlinks
+# the entire directory under Windows.
+test-%:
+	if [ -d "$*" ]; then \
+		echo "Running $*..." ; \
+		$(MAKE) -C "$*" test || exit 1; \
+	fi;
+
+# Run all test directories using GNU Parallel
+.PHONY: test-parallel $(DIRS)
+.NOTPARALLEL: test-parallel
+test-parallel:
+	@echo "Building with $(JOBS)"
+	parallel --halt soon,fail=1 \
+		--tag \
+		--tagstring '{#}:' \
+		--linebuffer \
+ 		--jobs $(JOBS) \
+ 		$(MAKE) "test-{}" \
+ 			::: $(DIRS)
```

Comment:
I'm not sure what the indentation tells here?

Reply:
It tells that @chrisr-diffblue can't type... :-)